server/test/auth.test.js

```diff
@@ -0,0 +1,89 @@
+require("dotenv").config();
+
+require("../models/User");
+const chai = require("chai");
+const chaiHttp = require("chai-http");
+const { app } = require("../app.js");
+const User = require("../models/User");
+
+const newUser = {
+  username: "testuser",
+  password: "testpassword",
+  email: "test@gmail.com",
+};
+
+chai.should();
+chai.use(chaiHttp);
+
+describe("Test for all auth endpoints", () => {
+  describe("POST /auth/register: Test for new user registeration", () => {
+    it("register should return 201 code, cookie token and logged in user", (pass) => {
+      chai
+        .request(app)
+        .post("/auth/register")
+        .send(newUser)
+        .end((err, res) => {
+          res.should.have.status(201);
+          res.should.have.cookie("token");
+          res.body.should.have.property("success");
+          res.body.success.should.have.property("user");
+          pass();
+        });
+    });
+
+    it("register should return 400 code when user already exists", (pass) => {
```

Comment:
`400` is probably not the most appropriate status code for this case (you can find some suggestions on what might be on this [thread](https://stackoverflow.com/questions/9269040/which-http-response-code-for-this-email-is-already-registered/9270432)). 

Ideally, we don't want to define success/failure of our test cases based on the implementation. Therefore I would recommend not looking at the implementation(s) of the feature(s) you're testing as much as possible while designing them (as if to pretend they aren't implemented yet) 👍🏼